### ISSUE TITLE
`.ghcup/env` file seems wrong in Windows

### ISSUE THREAD

**Issue by jasagredo** [haskell/ghcup-hs#955]
I don't think I provided any particular flags when installing GHCup (it was some time ago, I could delete it and install it again) but I think the env file is somewhat wrong. My installation folders are `/c/ghcup` and `/c/cabal`, however the script reads as follows:

```
➜ cat /c/ghcup/env
case ":$PATH:" in
    *:"/c/ghcup/bin":*)
        ;;
    *)
        export PATH="/c/ghcup/bin:$PATH"
        ;;
esac
case ":$PATH:" in
    *:"$HOME/.cabal/bin":*)
        ;;
    *)
        export PATH="$HOME/.cabal/bin:$PATH"
        ;;
esac
```

The `/c/ghcup` path is correct, but the `.cabal` one is not.

**Comment by jasagredo** [comment]
Maybe I installed `ghcup` inside mingw (therefore using `bootstrap-haskell`) instead of inside powershell (so not using `bootstrap-haskell.ps1`)? that would probably be an important fact to know.

**Comment by hasufell** [comment]
For the env script on windows we can assume a mingw environment. 

In the powershell environment, we rely on proper windows environment variables. 

Indeed, we seem to hardcode the location:

https://github.com/haskell/ghcup-hs/blob/524cdbbeb169d68a26dadb3a1111aaa8754e3c0a/scripts/bootstrap/bootstrap-haskell#L505

So it's sub-optimal. The problem is that cabal has differing defaults for the bin location, which depends on directory existence on your filesystem (bad implementation of the XDG spec: https://github.com/haskell/cabal/pull/7386).

And cabal has also no proper cli interface to discover those locations afaik. Maybe @fendor knows more.

**Comment by fendor** [comment]
Yeah, I don't think there is currently a way. In cabal head there is now `cabal path --installdir` (we want to rework the command as the latest cabal meeting), and could be usable in the future.
